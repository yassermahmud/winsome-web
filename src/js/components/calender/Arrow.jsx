import React from 'react';
import PropTypes from 'prop-types'
import {connect} from 'react-redux';
import {onClickRightArrow,onClickLeftArrow} from '../../actions'

export class Arrow extends React.Component{
	constructor()
	{
		super();
	}
	render(){
		let {onClickRightArrow,onClickLeftArrow}=this.props;
		return (
			<div>
				<div className="arrowRight"
				onClick={onClickRightArrow}
				>
				
				</div>
				<div className="arrowLeft"
				onClick={onClickLeftArrow}
				>
				</div>
			</div>
		)
	}
}
Arrow.propTypes={
	onClickRightArrow:PropTypes.func,
	onClickLeftArrow:PropTypes.func
}
export default connect(null,{onClickLeftArrow,onClickRightArrow})(Arrow);