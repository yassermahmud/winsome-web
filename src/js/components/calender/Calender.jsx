import React from 'react';
import Content from './Content';
import events from './Data';
import {connect} from 'react-redux';

export class Calender extends React.Component {
	constructor(){
		super();
	}
	render(){
		
		return (
			<div>
			<Content/>	
			</div>
		)
	}
}

export default connect(null,null)(Calender); 