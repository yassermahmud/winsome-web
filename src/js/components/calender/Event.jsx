import React from 'react'
import events from './Data'
import {connect} from 'react-redux'


export const Event=({counter})=>{
	let selectedEvent=[...events];
		return selectedEvent.slice(counter,counter+4).map(({...item},index)=>(
				<div key={index} className="event">
					<div className="comingEvents">
						{item['event']}
					</div>
					<div className="comingEventsDates">
							{item['eventDate']}
					</div>
					<div className="time">
							{item['time']}
					</div>
					<div className="clear"></div>
				</div>
				

		))
	}
function mapStateToProps({counter}) {
	return {
		counter
	}
}
export default connect(mapStateToProps,null)(Event)
