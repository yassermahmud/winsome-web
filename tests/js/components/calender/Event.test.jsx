import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Event} from '../../../../src/js/components/calender/Event';

Enzyme.configure({adapter:new Adapter()})

describe('Test the Event Component of Calender',()=>{
	const wrapper=shallow(<Event />);
	it('checks the rendring',()=>{
		expect(wrapper).toMatchSnapshot();
	})
	it('Test the return of array of objects',()=>{
		let items=[{event:'Half terms End',eventDate:'01 May',time:''},{},{}];
		expect(items[0]).toEqual({event:'Half terms End',eventDate:'01 May',time:''})
	})
})
