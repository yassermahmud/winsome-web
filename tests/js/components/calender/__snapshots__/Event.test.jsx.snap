// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Event Component of Calender checks the rendring 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Event />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [],
  Symbol(enzyme.__nodes__): Array [
    Array [],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
